src/main/scala/tech/cryptonomic/conseil/tezos/DatabaseConversions.scala

```diff
@@ -33,91 +36,85 @@ object DatabaseConversions {
   //implicit conversions to database row types
 
   implicit val averageFeesToFeeRow = new Conversion[Id, AverageFees, Tables.FeesRow] {
-    override def convert(from: AverageFees) =
-      Tables.FeesRow(
-        low = from.low,
-        medium = from.medium,
-        high = from.high,
-        timestamp = from.timestamp,
-        kind = from.kind
-      )
+    override def convert(from: AverageFees) = from.transformInto[Tables.FeesRow]
   }
 
   implicit val blockAccountsToAccountRows =
     new Conversion[List, BlockTagged[Map[AccountId, Account]], Tables.AccountsRow] {
       override def convert(from: BlockTagged[Map[AccountId, Account]]) = {
         val BlockTagged(hash, level, accounts) = from
-        accounts.map {
-          case (id, Account(manager, balance, spendable, delegate, script, counter)) =>
-            Tables.AccountsRow(
-              accountId = id.id,
-              blockId = hash.value,
-              manager = manager.value,
-              spendable = spendable,
-              delegateSetable = delegate.setable,
-              delegateValue = delegate.value.map(_.value),
-              counter = counter,
-              script = script.map(_.code.expression),
-              storage = script.map(_.storage.expression),
-              balance = balance,
-              blockLevel = level
-            )
-        }.toList
+
+        implicit val rowTransformer: Transformer[(AccountId, Account), Tables.AccountsRow] =
+          (entry: (AccountId, Account)) => {
+            val (id, acc) = entry
+            acc
+              .into[Tables.AccountsRow]
+              .withFieldConst(_.accountId, id.id)
+              .withFieldConst(_.blockId, hash.value)
+              .withFieldConst(_.blockLevel, BigDecimal(level))
+              .withFieldComputed(_.delegateSetable, _.delegate.setable)
```

Comment:
why `withFieldComputed`?

Reply:
It's reading inside an inner component, I don't believe chimney's able to figure that out...
In fact this exercise led me to think that multiple input patching could be really a huge addition to the library.
That is, you give me a, b, c and tell me how to use 'em all to build a d